### ISSUE TITLE
ls-remote should support 'engine'

### ISSUE THREAD

**Issue by mykhalov** [tj/n#674]
# Bug Report

`ls-remote` should support 'engine' option as the version as `install` does.

## Steps to Reproduce

1. Have a package.json with `"engine": { "node": "12" }`
2. Run `n ls-remote engine`

## Expected Behaviour

An output identical to `n ls-remote 12`

## Actual Behaviour

```
Error: invalid version 'engine'
```

## Configuration Details

```bash
$ n --version
7.2.2

$ command -v node
12.8.0

$ node -p process.platform
darwin
```

**Comment by shadowspawn** [comment]
Why do want to run `n ls-remote engine`? Are you running it yourself to see what version would be installed, or all matching versions, or perhaps wanting to use it in a script to find resolved version?

(I decided not to implement `ls-remote` at the time I added `auto` and `engine` as didn't seem a good fit and was extra work, so interested what you want to use it for.)

**Comment by mykhalov** [comment]
My use case is to print what exact version will be installed:

```
% n ls-remote 12 --all | head -1
12.22.1
```

Not being able to use `version` here feels like a bug and isn't mention in the docs:

```
% n --help

Usage: n [options] [COMMAND] [args]

Commands:

[...]
  n install <version>            Install Node.js <version> (downloading if necessary)
[...]
  n ls-remote [version]          Output matching versions available for download
[...]

Versions:
[...]
    engine            Read version from package.json
[...]
```

**Comment by shadowspawn** [comment]
Released in v7.3.0
